### PR TITLE
DEV-3670 Sundry improvements to Platinum

### PR DESCRIPTION
**Pull request by nedleitch**
This turned into a bit of a catch-all but I think the per-ticket commit messages are pretty useful.

### CODE REVIEW

File: src/main/java/com/hartwig/platinum/config/version/PipelineVersion.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        ArrayList<String> versionsGiven = new ArrayList<>(List.of(candidate.split("\\.")));
+        int i;
+        int numMissing = 3 - versionsGiven.size();
+        for (i = 0; i < numMissing; i++) {
+            versionsGiven.add("0");
+        }
+        return String.join(".", versionsGiven);
+    }
+
+    String extract(final String fullPipelineVersion) {
```
Comment (KasperWolsink99) [inline src/main/java/com/hartwig/platinum/config/version/PipelineVersion.java:19]: Maybe rename method to something slightly specific IE `extractVersionFromDockerImageName`

File: src/main/java/com/hartwig/platinum/config/version/PipelineVersion.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import static java.lang.String.format;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PipelineVersion {
+    String pad(final String candidate) {
+        ArrayList<String> versionsGiven = new ArrayList<>(List.of(candidate.split("\\.")));
+        int i;
+        int numMissing = 3 - versionsGiven.size();
```
Comment (KasperWolsink99) [inline src/main/java/com/hartwig/platinum/config/version/PipelineVersion.java:12]: I think this method should throw if `versionsGiven.size() > 3`

**Review by KasperWolsink99 — APPROVED** [review]
Approved with some slight improvement suggestions

File: src/main/java/com/hartwig/platinum/config/version/PipelineVersion.java (hunk 2 above)
Comment (nedleitch) [reply src/main/java/com/hartwig/platinum/config/version/PipelineVersion.java:15]: Agreed. Corrected.

File: src/main/java/com/hartwig/platinum/config/version/PipelineVersion.java (hunk 1 above)
Comment (nedleitch) [reply src/main/java/com/hartwig/platinum/config/version/PipelineVersion.java]: Nice suggestion. Corrected.

**Pull request merged**